src/lightning_lite/cli.py

```diff
@@ -0,0 +1,151 @@
+            " 0, ..., num_nodes - 1."
+        ),
+    )
+    parser.add_argument(
+        "--main-address",
+        type=str,
+        default="127.0.0.1",
+        help="The hostname or IP address of the main machine (usually the one with node_rank = 0).",
+    )
+    parser.add_argument(
+        "--main-port",
+        type=int,
+        default=29400,
+        help="The main port to connect to the main machine.",
+    )
+    parser.add_argument(
+        "--precision",
+        type=str,
+        default="32",
+        choices=_SUPPORTED_PRECISION,
+        help=(
+            "Double precision (``64``), full precision (``32``), half precision (``16``) or bfloat16 precision"
+            " (``'bf16'``)"
+        ),
+    )
+
+    args, script_args = parser.parse_known_args()
+    return args, script_args
+
+
+def _set_env_variables(args: Namespace) -> None:
+    """Set the environment variables for the new processes.
+
+    The Lite connector will parse the arguments set here.
+    """
+    os.environ["LT_CLI_USED"] = "1"
+    os.environ["LT_ACCELERATOR"] = str(args.accelerator)
+    if args.strategy is not None:
+        os.environ["LT_STRATEGY"] = str(args.strategy)
+    os.environ["LT_DEVICES"] = str(args.devices)
+    os.environ["LT_NUM_NODES"] = str(args.num_nodes)
+    os.environ["LT_PRECISION"] = str(args.precision)
+
+
+def _get_num_processes(accelerator: str, devices: str) -> int:
+    """Parse the `devices` argument to determine how many processes need to be launched on the current machine."""
+    if accelerator in ("cuda", "gpu"):
+        parsed_devices = CUDAAccelerator.parse_devices(devices)
+    elif accelerator in ("mps", "gpu"):
+        parsed_devices = MPSAccelerator.parse_devices(devices)
+    elif accelerator == "tpu":
+        raise ValueError("Launching processes for TPU through the CLI is not supported.")
+    else:
+        return CPUAccelerator.parse_devices(devices)
+    return len(parsed_devices) if parsed_devices is not None else 0
+
+
+def _torchrun_launch(args: Namespace, script_args: List[str]) -> None:
+    """This will invoke `torchrun` programmatically to launch the given script in new processes."""
+    if args.strategy == "dp":
+        num_processes = 1
+    else:
+        num_processes = _get_num_processes(args.accelerator, args.devices)
+
+    torchrun_args = []
+    torchrun_args.extend(["--nproc_per_node", str(num_processes)])
+    torchrun_args.extend(["--nnodes", str(args.num_nodes)])
+    torchrun_args.extend(["--node_rank", str(args.node_rank)])
+    torchrun_args.extend(["--master_addr", args.main_address])
+    torchrun_args.extend(["--master_port", str(args.main_port)])
+    torchrun_args.append(args.script)
+    torchrun_args.extend(script_args)
+
+    # set a good default number of threads for OMP to avoid warnings being emitted to the user
+    os.environ.setdefault("OMP_NUM_THREADS", str(max(1, (os.cpu_count() or 1) // num_processes)))
```

Comment:
Why do we need `max`?

```suggestion
    os.environ.setdefault("OMP_NUM_THREADS", str((os.cpu_count() or 1) // num_processes))
```

Reply:
The division with num_processes can lead to a number smaller than 1, which gets floored to 0. max(1, x) clips the value to >=1, so OMP_NUM_THREADS is at least 1.